microprofile/grpc/client/src/main/java/io/helidon/microprofile/grpc/client/model/ClientServiceModeller.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.grpc.client.model;
+
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import io.helidon.grpc.client.ClientMethodDescriptor;
+import io.helidon.grpc.client.ClientServiceDescriptor;
+import io.helidon.grpc.core.MethodHandler;
+import io.helidon.microprofile.grpc.core.GrpcMarshaller;
+import io.helidon.microprofile.grpc.core.RpcMethod;
+import io.helidon.microprofile.grpc.core.model.AbstractServiceModeller;
+import io.helidon.microprofile.grpc.core.model.AnnotatedMethod;
+import io.helidon.microprofile.grpc.core.model.AnnotatedMethodList;
+import io.helidon.microprofile.grpc.core.model.Instance;
+import io.helidon.microprofile.grpc.core.model.ModelHelper;
+
+/**
+ * Utility class for constructing a {@link ClientServiceDescriptor.Builder}
+ * from an annotated POJO.
+ */
+public class ClientServiceModeller
+        extends AbstractServiceModeller {
+
+    private static final Logger LOGGER = Logger.getLogger(ClientServiceModeller.class.getName());
+
+    /**
+     * Create a new introspection modeller for a given gRPC service.
+     *
+     * @param service the service to call gRPC handler methods on
+     * @throws NullPointerException if the service is null
+     */
+    public ClientServiceModeller(Object service) {
+        this(service.getClass(), Instance.singleton(service));
```

Comment:
Please never use public contructors anywhere in Helidon (except for very specific cases - see dev guidelines).
If the instance created is the final instance needed, then this should be handled by static factory methods:
`static create(Object)`
`static create(Class)`
`static create(Class, Supplier)`
If these are needed to create a builder, then these should be parameters to a static builder method:
`static builder(Object)`
`static builder(Class)`
`static builder(Class, Supplier)`